projects/bp-gallery/src/views/gallery/search/helpers/search-filters.ts

```diff
@@ -0,0 +1,234 @@
+import { PictureFiltersInput } from '../../../../graphql/APIConnector';
+import { SearchType } from '../SearchView';
+
+const paramToTimefilter = (timeParam: string) => {
+  if (timeParam === 'pre50') timeParam = '49';
+  const year = parseInt(timeParam);
+
+  if (!isNaN(year)) {
+    let startTime = `${year}-01-01T00:00:00Z`;
+    let endTime = `${year}-12-31T23:59:59Z`;
+    if (year < 100) {
+      startTime = year === 49 ? '1900-01-01T00:00:00Z' : `19${year}-01-01T00:00:00Z`;
+      endTime = `19${year}-12-31T23:59:59Z`;
+    }
+
+    const time_range_tag_filter = {
+      start: {
+        gte: startTime,
+      },
+      end: {
+        lte: endTime,
+      },
+    };
+
+    return time_range_tag_filter;
+  }
+};
+
+const searchDecade = (searchParams: URLSearchParams, filters: PictureFiltersInput) => {
```

Comment:
I think it should work as follows:
* When searching for a year via the search bar (all or year only), you should only filter for pictures from that year
* When using a decade suggestion, you should find all images from that decade (19X0 - 19X9)